starfyre/parser.py

```diff
@@ -150,25 +158,40 @@ def handle_endtag(self, tag):
         # if the tag is not found in the generic tags and custom components
         if tag not in self.generic_tags and tag not in self.components:             
             raise UnknownTagError(f'Unknown tag: "{tag}". Please review line {self.lineno} in your "{self.component_name}" component in the pyml code.')
-
-        # we need to check if the tag is a default component or a custom component
-        # if it is a custom component, we get the element from the custom components dict          
-
-        if tag not in self.generic_tags and tag in self.components:
-            component = self.components[tag]
-            tag = component.tag
         
         endtag_node = self.stack.pop()  
         self.current_depth -= 1
+
         if endtag_node.tag != "style" and endtag_node.tag != "script":
             if len(self.stack) > 0:
                 parent_node = self.stack[-1]      #this is last item/"top element" of stack
-                parent_node.children.append(endtag_node)
+
+                if endtag_node.original_name != endtag_node.tag: #this node is for "custom" component
+                    #process endtag_node children                    
+                    new_children = []
+                    is_slot_used = False
+                    for child_component in endtag_node.children:                        
+                        if child_component.is_slot_component:
+                            new_children.extend(self.current_children)
+                            is_slot_used = True
+                        else:
+                            new_children.append(child_component)
+                    if not is_slot_used and len(self.current_children) > 0:
+                        new_children.extend(self.current_children)
+                        print("Appending at the end of the stack as slot position not specified.")
+                    endtag_node.children = new_children   
+                    self.current_children = []                     
+
+                if parent_node.original_name != parent_node.tag:
+                    self.current_children.append(endtag_node)
+                else:
+                    parent_node.children.append(endtag_node)    
             else:
                 self.root_node = endtag_node
-      
+                self.root_node.children.extend(self.current_children)
+                self.current_children = []
```

Comment:
This part needs to be reviewed further and will require some changes. @SuelenKarbivnychyy , can you please add the test files?

Reply:
> This part needs to be reviewed further and will require some changes. @SuelenKarbivnychyy , can you please add the test files?

@sansyrox 

I'm using `parent.fyre` to test it. `<slot> tag` is being using on line 20 of that file.

Reply:
> This part needs to be reviewed further and will require some changes. @SuelenKarbivnychyy , can you please add the test files?

@sanskar

What are the changes required to make it ready to merge?